api/reporting/schemas/reporting-codegen.rb

```diff
@@ -26,64 +27,187 @@ def to_output_path(dir_name, table_name, suffix)
   File.expand_path(File.join(dir_name, "#{table_name}.#{suffix}"))
 end
 
-dto_class_name = "BqDto#{to_camel_case(table_name, true)}"
-
-inputs = {
+INPUTS = {
     :describe_csv => to_input_path(File.join(input_dir, 'mysql_describe_csv'), table_name,'csv'),
-    :exclude_columns => to_input_path(File.join(input_dir, 'excluded_columns'), table_name,'txt')
-}
+    :excluded_columns => to_input_path(File.join(input_dir, 'excluded_COLUMNS'), table_name,'txt')
+}.freeze
 
-outputs = {
+OUTPUTS = {
```

Comment:
There's a lot going on in this script. Where is the package-level description of what this tool is doing? Details like inputs, outputs, and how Swagger fits in.

In terms of usage of the tool: is this meant to be run once and then likely never again? Or are we expecting to run this continually as the database and/or reporting schemas evolve? If so - is there a reproducible mechanism for doing so? I'm noticing seemingly a lot of manual modifications post-codegen, which would make this challenging.

Reply:
The short answer is that I needed it to keep things consistent, and would like to re-use it for schema modifications. There are still a couple of special cases, but I was prioritizing getting somethign shipped on the reporting side.

For this to be a long-lived, clean tool, I'd probably want to port it to use a legitimate templating system, and have it occur at build time rather than just once.

Reply:
There's this [README](https://github.com/all-of-us/workbench/blob/13c3ffd798c51d84fe61117fe3707743c7ca044f/api/reporting/schemas/REPORTING-SCHEMA-TOOLS.md) for the whole thing, which would need to be updated if we stick with this structure. I'm scrambling to get a dataset delivered this week, so haven't gotten to update it  yet.